shell/browser/electron_browser_client.cc

```diff
@@ -1313,6 +1315,15 @@ bool ElectronBrowserClient::WillInterceptWebSocket(
   if (!web_request.get())
     return false;
 
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+  const auto* web_request_api =
+      extensions::BrowserContextKeyedAPIFactory<extensions::WebRequestAPI>::Get(
+          browser_context);
+
+  if (web_request_api)
+    return web_request->HasListener() || web_request_api->MayHaveProxies();
```

Comment:
```suggestion
  if (web_request_api)
    return web_request->HasListener() || web_request_api->MayHaveProxies() || web_request->HasListener();
```

?

It's possible that the `WebRequestAPI` is available but doesn't have a listener, but electron's `api::WebRequest` does have a listener.

Reply:
oh nevermind i'm blind